performance_test/helper_scripts/regression_checkers/ApexComparison.py

```diff
@@ -0,0 +1,554 @@
+                # Check if value is print_results
+                elif argument == valid_args[3]:
+                    # Check if value is boolean
+                    try:
+                        print_results = bool_cast(arguments[i + 1])
+                        i += 1
+                    except ValueError:
+                        logger.error('Incorrect argument in {}'.format(
+                            valid_args[3])
+                        )
+                        return None
+            else:
+                return None
+
+            # Increase loop counter
+            i += 1
+
+    ret = {
+        'script_name': script_name,
+        'ref_dir': ref_dir,
+        'target_dir': target_dir,
+        'latency_threshold': latency_threshold,
+        'rss_threshold': rss_threshold,
+        'jitter_threshold': jitter_threshold,
+        'print_results': print_results,
+    }
+
+    return ret
+
+
+def get_column_ids():
+    """Get a list of log file column ids."""
+    columns_ids = [
+        'T_experiment', 'T_loop', 'received', 'sent', 'lost', 'relative_loss',
+        'data_received', 'latency_min (ms)', 'latency_max (ms)',
+        'latency_mean (ms)', 'latency_variance (ms)', 'pub_loop_res_min (ms)',
+        'pub_loop_res_max (ms)', 'pub_loop_res_mean (ms)',
+        'pub_loop_res_variance (ms)', 'sub_loop_res_min (ms)',
+        'sub_loop_res_max (ms)', 'sub_loop_res_mean (ms)',
+        'sub_loop_res_variance (ms)', 'ru_utime', 'ru_stime', 'ru_maxrss',
+        'ru_ixrss', 'ru_idrss', 'ru_isrss', 'ru_minflt', 'ru_majflt',
+        'ru_nswap', 'ru_inblock', 'ru_oublock', 'ru_msgsnd', 'ru_msgrcv',
+        'ru_nsignals', 'ru_nvcsw', 'ru_nivcsw'
+    ]
+    return columns_ids
+
+
+def get_subdirectories(rates, num_subs):
+    """
+    Get the list of expected subdirectories in every directory.
+
+    The expected subdirectories are obtaines from the Cartesian product
+    of the elements of rates and num_subs.
+
+    :param rates (list): The sending rates
+    :param num_subs (list): All the different number of subscribers.
+
+    :return:
+        A list containing the path of all the subdirectories.
+
+    """
+    # Get subdirectory names
+    combinations = list(itertools.product(rates, num_subs))
+    sub_dirs = []
+    for combination in combinations:
+        sub_dirs.append('rate_{}/subs_{}'.format(
+            combination[0], combination[1])
+        )
+    return sub_dirs
+
+
+def get_file_names(directory, sub_dirs):
+    """
+    Get the file name of all the files in every subdirectory of directory.
+
+    :param directory (str): The main directory.
+    :param sub_dirs (list): A list of subdirectories to look for files.
+
+    :return: A list of file names in the form:
+        'directory/subdirectory/filename'
+    """
+    files = {}
+    for sub_dir in sub_dirs:
+        dir_name = '{}/{}'.format(directory, sub_dir)
+
+        files_l = [f for f in listdir(dir_name) if isfile(join(dir_name, f))]
```

Comment:
You can also call `sorted([...])` here instead.

Reply:
Fixed